src/seedu/addressbook/data/person/Address.java

```diff
@@ -22,11 +30,21 @@
      */
     public Address(String address, boolean isPrivate) throws IllegalValueException {
         String trimmedAddress = address.trim();
-        this.isPrivate = isPrivate;
-        if (!isValidAddress(trimmedAddress)) {
+        Pattern parserPattern = Pattern.compile(ADDRESS_VALIDATION_REGEX);
+        Matcher parserMatcher = parserPattern.matcher(trimmedAddress);
+   
+        if (parserMatcher.find()) {
+        	_block = new Block(parserMatcher.group(1));
+        	_street = new Street(parserMatcher.group(3));
+        	_unit = new Unit(parserMatcher.group(5));
+        	_postalCode = new PostalCode(parserMatcher.group(7));
+        }
+        
+        else {
             throw new IllegalValueException(MESSAGE_ADDRESS_CONSTRAINTS);
         }
-        this.value = trimmedAddress;
+        
+        value = _block.getValue() + ", " +  _street.getValue() + ", " + _unit.getValue() + ", " + _postalCode.getValue();
```

Comment:
No need to save the input address in `value` as it may cause inconsistency due to duplication.